redbot/cogs/mod/kickban.py

```diff
@@ -210,15 +210,19 @@ class KickBanMixin(MixinMeta):
         self,
         ctx: commands.Context,
         user: discord.Member,
-        days: Optional[int] = 0,
+        days: Optional[int] = None,
         *,
         reason: str = None,
     ):
         """Ban a user from this server and optionally delete days of messages.
 
         If days is not a number, it's treated as the first word of the reason.
         Minimum 0 days, maximum 7. Defaults to 0."""
-
+        guild = ctx.guild
+        if days == None:
+            days = await self.settings.guild(guild).default_days()
+        if days == 0:
```

Comment:
This is pointless, you check if it's equal to 0 and then set it to 0 if it is

Reply:
While I had it as if days ==None:
if 0 was said in the command for days it was still taking the default days, although after changing if days == None: to if days is None: fixed that and I've removed that part.